frontend/public/components/storage/create-pvc.tsx

```diff
@@ -38,357 +38,313 @@ const provisionerAccessModeMapping = {
-      {
-        value: 'ReadWriteMany',
-        title: 'Shared Access (RWX)',
-      },
-      {
-        value: 'ReadOnlyMany',
-        title: 'Read Only (ROX)',
-      },
-    ],
-    dropdownUnits: {
-      Mi: 'Mi',
-      Gi: 'Gi',
-      Ti: 'Ti',
+export const CreatePVCForm: React.FC<CreatePVCFormProps> = (props) => {
+  const [accessModeHelp, setAccessModeHelp] = React.useState('Permissions to the mounted drive.');
+  const [allowedAccessModes, setAllowedAccessModes] = React.useState([
+    'ReadWriteOnce',
+    'ReadWriteMany',
+    'ReadOnlyMany',
+  ]);
+  const [storageClass, setStorageClass] = React.useState('');
+  const [pvcName, setPvcName] = React.useState('');
+  const [accessMode, setAccessMode] = React.useState('ReadWriteOnce');
+  const [requestSizeValue, setRequestSizeValue] = React.useState('');
+  const [requestSizeUnit, setRequestSizeUnit] = React.useState('Gi');
+  const [useSelector, setUseSelector] = React.useState(false);
+  const [nameValuePairs, setNameValuePairs] = React.useState([['', '']]);
+  const accessModeRadios = [
+    {
+      value: 'ReadWriteOnce',
+      title: 'Single User (RWO)',
+    },
+    {
+      value: 'ReadWriteMany',
+      title: 'Shared Access (RWX)',
     },
+    {
+      value: 'ReadOnlyMany',
+      title: 'Read Only (ROX)',
+    },
+  ];
+  const dropdownUnits = {
+    Mi: 'Mi',
+    Gi: 'Gi',
+    Ti: 'Ti',
   };
+  const { namespace, onChange } = props;
+
+  React.useEffect(() => {
+    const getSelector = () => {
+      if (!useSelector) {
+        return null;
+      }
+
+      const matchLabels = _.reduce(
+        nameValuePairs,
+        (acc, [key, value]) => {
+          return key ? { ...acc, [key]: value } : acc;
+        },
+        {},
+      );
 
-  handleChange: React.ReactEventHandler<HTMLInputElement> = (event) => {
-    // this handles pvcName, accessMode, size
-    const { name, value } = event.currentTarget;
-    this.setState({ [name]: value } as any, this.onChange);
-  };
+      return _.isEmpty(matchLabels) ? null : { matchLabels };
+    };
+    const updatePVC = (): K8sResourceKind => {
+      const obj: K8sResourceKind = {
+        apiVersion: 'v1',
+        kind: 'PersistentVolumeClaim',
+        metadata: {
+          name: pvcName,
+          namespace,
+        },
+        spec: {
+          accessModes: [accessMode],
+          resources: {
+            requests: {
+              storage: `${requestSizeValue}${requestSizeUnit}`,
+            },
+          },
+        },
+      };
+
+      // Add the selector only if specified.
+      const selector = getSelector();
+      if (selector) {
+        obj.spec.selector = selector;
+      }
 
-  handleNameValuePairs = ({ nameValuePairs }) => {
-    this.setState({ nameValuePairs }, this.onChange);
+      if (storageClass) {
+        obj.spec.storageClassName = storageClass;
+      }
+
+      return obj;
+    };
+    onChange(updatePVC);
+  }, [
+    accessMode,
+    accessModeHelp,
+    allowedAccessModes,
```

Comment:
I don't think this needs to be a useEffect dependency

Reply:
Cleaned that up.